test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,177 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	_ "embed"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	e2e "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+)
+
+// LinuxVMFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type LinuxVMFakeintakeSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+//go:embed log-config/log-config.yaml
+var logConfig []byte
+
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef() *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+
+	return e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", string(logConfig))))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	s := &LinuxVMFakeintakeSuite{}
+	if _, devmode := os.LookupEnv("TESTS_E2E_DEVMODE"); devmode {
+		s.DevMode = true
+	}
+
+	e2e.Run(t, &LinuxVMFakeintakeSuite{}, logsExampleStackDef())
+}
+
+func (s *LinuxVMFakeintakeSuite) BeforeTest(_, _ string) {
+	// Flush server and reset aggregators before the test is ran
+	if s.DevMode {
+		s.cleanUp()
+	}
+	s.Env().Fakeintake.FlushServerAndResetAggregators()
+}
+
+func (s *LinuxVMFakeintakeSuite) TearDownSuite() {
+	// Flush server and reset aggregators after the test is ran
+	if s.DevMode {
+		s.cleanUp()
+	}
+	s.Env().Fakeintake.FlushServerAndResetAggregators()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should this be part of `s.cleanUp()` ?